astropy/coordinates/matrix_utilities.py

```diff
@@ -88,7 +90,7 @@ def rotation_matrix(angle, axis='z', unit=None):
     else:
         a1 = (i + 1) % 3
         a2 = (i + 2) % 3
-        R = np.zeros(angle.shape + (3, 3))
+        R = np.zeros(np.shape(angle) + (3, 3))
```

Comment:
If we're micro-optimizing anyway, use `getattr(angle, 'shape', ()` (58 ns instead of 1.73 us for a scalar)

Reply:
I love this optimization

Reply:
Or non-optimization ?
```
In [21]: %timeit np.shape(a)       
326 ns ± 0.521 ns per loop (mean ± std. dev. of 7 runs, 1000000 loops each)

In [22]: %timeit a.shape           
23.2 ns ± 0.0369 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [23]: %timeit getattr(a, 'shape', ())                              
50.5 ns ± 0.69 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
@saimn The above tests do not include the rest of the code. For example `a.shape` will not work if `a = 15` (i.e., degrees). For the most typical use when the angle is a scalar, @mhvk optimization is great. For the same reason I switched to `np.shape()` so that if input angles are in radians and provided such as `(angle=[1, 2], unit=np.rad)` the angles can be used "as is".

Reply:
@saimn - indeed, the reason this became necessary is that now scalars can go all the way down to here. I think the `getattr` way is the fastest one to deal with both cases.

Reply:
Ah ok, I missed the scalar context ;).
But still I'm not sure 200ns are worth the complexity added by this `getattr`.

Reply:
Ok, if `a` is a Python scalar I get the same 1us as @mhvk .